Source/ASCollectionView.mm

```diff
@@ -960,30 +960,38 @@ - (CGSize)collectionView:(UICollectionView *)collectionView layout:(UICollection
 {
   ASDisplayNodeAssertMainThread();
   NSIndexPath *indexPath = [NSIndexPath indexPathForItem:0 inSection:section];
-  ASCellNode *cell = [self supplementaryNodeForElementKind:UICollectionElementKindSectionHeader
-                                               atIndexPath:indexPath];
-  if (cell.shouldUseUIKitCell && _asyncDelegateFlags.interop) {
+  ASCollectionElement *element = [_dataController.visibleMap supplementaryElementOfKind:UICollectionElementKindSectionHeader
```

Comment:
Because `-supplementaryNodeForElementKind:atIndexPath` internally calls `[_dataController.visibleMap supplementaryElementOfKind:atIndexPath].node`, we're literally asking for the element twice in this method. Let's directly reach out to the visible map.